Update release strategy notes (#2734)
ESMValGroup/ESMValTool: doc/sphinx/source/community/release_strategy.rst

   - Any contributor is welcome to add issues or pull requests that they intend to work on themselves to a milestone.

> #. ESMValCore feature freeze, testing and release

Reviewer: ```suggestion
#. ESMValCore feature freeze, testing, and release
```

   - A release branch is created and branch protection rules are set up so only the release manager (i.e. the person in charge of the release branch) can push commits to that branch.
   - Make a release candidate with the release branch. Note that the :ref:`ESMValCore release instructions <esmvalcore:how-to-make-a-release>` are currently outdated and pending to be modified,

>      but they are analogous to the ESMValTool instructions listed in :ref: `How to make a release`. 

Reviewer: The link is not shown correctly on readthedocs, could you have a look?

Author: I think this line can be removed anyway

>    - Run all the recipes (optionally with a reduced amount of data) to check that they still work with the release candidate

Reviewer: This is explained in much more detail in step 4. Could you have a look if it still makes sense to have step 4 where it is now? Do we need new/different steps?

   - If a bug is discovered that needs to be fixed before the release, a pull request can be made to the main branch to fix the bug. The person making the pull request can then ask the release manager to cherry-pick that commit into the release branch.

>    - Make another release candidate with the discovered bugfixes and run again all recipes to check again for bugs.

Reviewer: ```suggestion
   - Make another release candidate with the bugfixe(s) and run the affected recipes again to check for further bugs.
```
Running all recipes again seems a lot of work if a bug only affected some recipes. Is this really needed?

>    - Make as many release candidates for ESMValCore candidates as needed in order to fix all the detected bugs.

Reviewer: ```suggestion
   - Make as many release candidates for ESMValCore as needed in order to fix all the detected bugs.
```

static version number is stored in ``CITATION.cff``.
Make sure to update the version number and release date in ``CITATION.cff``.
See https://semver.org for more information on choosing a version number.

> Update the `environment.yml <https://github.com/ESMValGroup/ESMValTool/blob/main/environment.yml#L25>`__

Reviewer: It seems that setup.py is missing here, but maybe you could just link to the section on [updating dependencies](https://esmvaltool--2734.org.readthedocs.build/en/2734/community/code_documentation.html#dependencies) to avoid duplicating information? That would also help to keep the information up to date if things change again in the future.

Author: Changed

Reviewer: Thanks

`ESMValTool development team <https://github.com/orgs/ESMValGroup/teams/esmvaltool-developmentteam>`__
to run their favourite recipe using this branch.

> 5. Cherry pick bugfixes into the release branch

Reviewer: I noticed that there are still mentions of cherry-picking in other places in this document. Would it make sense to update those too?

Reviewer: I noticed that you thought it was better to stick with the feature freeze and cherry-picking bug fixes into the release branch here: https://github.com/ESMValGroup/ESMValCore/pull/1689#discussion_r947650781, instead of allowing an endless stream of updates into the release candidate. I also think this is a good idea (that's why it is in the text as it is now), but this was done differently the past few releases.

@ESMValGroup/technical-lead-development-team Is this something we need to talk about at one of our upcoming meetings so we have a shared understanding of how we do releases?

Reviewer: Yes, but that advice is contrary to having a feature freeze unless the feature freeze means that only bug fixes can be merged into the `main` branch while the release is in progress. Let's talk about this with the team next week or at the next tech lead meeting.

Changelog
---------

> - 2022-07-21 Update instructions to follow the new release procedure

Reviewer 2: Not sure if this is necessary - the previous comment says "future changes tracked by git".

Author: Sure, I must have missed that comment
